src/resources/listing.js

```diff
@@ -0,0 +1,226 @@
+import Ajv from 'ajv'
+import URL from 'url-parse'
+import Money from '../utils/money'
+
+import schemaV1 from '../schemas/listing-core.json'
+
+const SCHEMA_V1 = '1.0.0'
+const schemaIdV1 = 'http://schema.originprotocol.com/listing-core-v1.0.0'
+
+const ajv = new Ajv({allErrors: true})
+// To use the draft-06 JSON schema, we need to explicitly add it to ajv.
+ajv.addMetaSchema(require('ajv/lib/refs/json-schema-draft-06.json'))
+ajv.addSchema(schemaV1)
+
+
+//
+// Listing is the main interface exposed by Origin Protocol to access data from a listing.
+//
+class Listing {
```

Comment:
I think this looks great, but with our current setup resources are publicly exposed. I actually think we do need to have a listing resource so that people can do things like `origin.listings.create`, etc. But that resource should contain the public methods.

If something is not public facing, we've been making it a `service`. I think making this a `listing-service` might make sense here, unless you have any other suggestions.

Reply:
One other thought: `service` doesn't necessarily make sense here to me either. This seems like a different type of thing than our contract service, for example.

What about `model`? We could start introducing the concept of models into this codebase. So this would be under `src/models/listing-model.js`